### PR TITLE
Add first draft of an “Architectural overview”

### PR DESCRIPTION
**Pull request by JohnSundell**
This document is aimed at developers who are looking to contribute to the Hub Framework, and want to get a more in-depth understanding of how the internals work, and what patterns that were chosen (and why).

The document is currently incomplete, but still offers some value (so pushing it up now for handover purposes). More work needs to be done to cover all aspects of the framework (with “Hubs sessions” and the other documentation being complementary to this document).

### CODE REVIEW

**Comment by JohnSundell** [comment]
@spotify/objc-dev

**Comment by rastersize** [comment]
I’ve made some tweaks and changes. I think it’s good enough to merge in this state giving us, and others, something to expand on.

File: documentation/Architectual overview.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+For information on how to use the framework see the other [available guides](https://spotify.github.io/HubFramework/index.html). Each guide introduces you to the various aspects of the framework, its APIs and [concepts](https://spotify.github.io/HubFramework/concept-guide.html).
+
+|  ℹ️  | _In progress_ |  ℹ️  |
+| ----- | :-----------: | ----- |
+| ℹ️ | **This docment is currently “work-in-progress” and does not yet offer a complete picture of the Hub Framework architecture.** | ℹ️ |
+
+## `HUBManager`
+
+`HUBManager` represents an insance of the Hub Framework. It’s not a singleton, but rather a core object that an application using the framework creates and holds a reference to. The manager then provides access to all of the framework’s features in a modular fashion.
```
Comment (mhallendal) [inline documentation/Architectual overview.md]: Typo on `instance` (missing t).

**Comment by JohnSundell** [comment]
Very nice additions @rastersize! 👍

**Comment by mhallendal** [comment]
👍

**Pull request merged**